metrics/ci_integrations/test/firestore/deserializer/build_data_deserializer_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:ci_integration/firestore/deserializer/build_data_deserializer.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('BuildDataDeserializer', () {
+    const id = 'id';
+    final buildData = BuildData(
+      id: id,
+      duration: const Duration(milliseconds: 100000),
+      startedAt: DateTime.now(),
+      url: 'testUrl',
+      buildNumber: 1,
+      buildStatus: BuildStatus.failed,
+      workflowName: 'testWorkflowName',
+      coverage: const Percent(1.0),
+    );
+
+    test(
+      '.fromJson() should throw NoSuchMethodError if a given json is null',
+      () {
+        expect(
+          () => BuildDataDeserializer.fromJson(null, id),
+          throwsNoSuchMethodError,
+        );
+      },
+    );
+
+    test('.fromJson() should return BuildData from a json map', () {
+      expect(
+        BuildDataDeserializer.fromJson(buildData.toJson(), id),
+        equals(buildData),
```

Comment:
I think we should create a separate JSON map and deserialize it instead of using `buildData.toJson()` since the real case will be deserializing a map from Firestore and not from the BuildData instance itself. Let me know what you think.

Reply:
Great job @vladlaba - I started reviewing this but will probably just wait for your comments resolved - as they covered a couple of my initial suggestions.